example/dummy/dummy.go

```diff
@@ -53,6 +54,18 @@ func (app *DummyApplication) Query(query []byte) types.Result {
 	return types.NewResultOK(wire.JSONBytes(queryResult), "")
 }
 
+func (app *DummyApplication) Proof(key []byte, blockHeight int64) types.Result {
+	if blockHeight != 0 {
```

Comment:
Well, this should be documented better.  Please suggest where.... (README.md?)

From the discussion with Jae, he wanted it to be possible to query for eg. 20 proofs all from the same (recent) block.  Due to timing, these 20 queries might end up with proofs from 2 or 3 different blocks. So, there is an optional blockHeight argument to specify the desired height (and 0 for whatever is latest).

Unfortunately, querying proofs from older blocks is currently impossible with go-merkle (related to the immutability conversation last week). So, currently only 0 is supported.  But in some time (< 3 months?), we should be able to maintain the last eg. 100 blocks, and allow querying proofs on any of them.

Thus, rather than rework it later and break all apps again, I added the semi-useless flag right now.